Working with embedded records

dustinfarris opened the pull request:
This creates blueprints that allow the user to extend the adapter/serializer, and adds documentation explaining how to use these blueprints when working with embedded records.

benkonrath commented on docs/extending.md:
```diff
@@ -0,0 +1,19 @@
+
+This will create app/serializers/user.js for you to customize.
+
+Similarly, if you wanted to, say, extend the adapter on the application level:
```
`... to, say, extend the adapter ...`

Suggestion: I think it's clearer if you use 'for example' instead of 'say'. Or even just leave 'say' out entirely.

benkonrath commented on docs/usage.md:
```diff
@@ -0,0 +1,99 @@
+# Usage
```
Are you planning that we should add other topics to this usage section? It's inconsistent if the pagination docs are separate but the embedded docs are in this usage page. I think that the embedded docs should be in their own sections and we can move the 'Path Customization' and 'coalesceFindRequests option' docs from the README to the usage section.

benkonrath commented on docs/extending.md:
```diff
@@ -0,0 +1,19 @@
+# Extending the adapter/serializer
```
The extending section from the README needs to be remove because it's no longer accurate.

dustinfarris replied:
Good point.  I removed everything usage-related from the README—we can keep all that in the docs going forward.

I did not copy the instructions to change how URLs are transformed.  This is already documented in the Ember docs, pretty easy to figure out with the blueprints, and I can't imagine it being something people will actually do too often.  If it becomes a pain point, we can add it back somewhere.

I also didn't copy the warning about the coalesceFindRequests.  It is disabled by default in ED anyway, and I plan on tackling it very soon.

benkonrath replied:
Seems reasonable.

benkonrath commented:
Great work! Just those small suggestions and then it's good to merge.

This is breaking change, right? I guess the change log should be updated as well.

benkonrath commented:
We should also add a note about not supporting writable nested relationships because DRF doesn't support it out of the box. 

http://www.django-rest-framework.org/topics/3.0-announcement/#writable-nested-serialization

I suppose something could be worked out in the future but other issue / features have a high priority right now (at least for me).

dustinfarris commented:
I think I addressed everything.

I don't think this is a breaking change.  A lot of stuff was moved around, but I can't think of a scenario where previous usage would be affected.

I added the note about not supporting writable nested relationships.  I agree that it is not a high priority (if it is even possible—does Ember actually allow this?)

benkonrath commented:
I thought it was  breaking change because generate commands have change. Or was this not working with 0.5.0? For example, you're removing this from the README:

>ember generate django-adapter my-custom-adapter

In any case, we can always fix up the changelog before the release if it needs fixing. Great work here.

dustinfarris commented:
Thanks!  Yeah, the instructions in the README were from < 0.5.0, just never got cleaned out when we replaced master.
